### ISSUE TITLE
System.ArgumentException occurred, width is less than zero, Avalonia.Win32

### ISSUE THREAD

**Issue by wieslawsoltes** [AvaloniaUI/Avalonia#1023]
Exception is thrown when using Avalonia 0.5.1 with Win32/Skia (Desktop and .Net Core projects):
```
System.ArgumentException occurred
  HResult=0x80070057
  Message=width is less than zero
  Source=Avalonia.Win32
  StackTrace:
   at Avalonia.Win32.WindowFramebuffer..ctor(IntPtr handle, Int32 width, Int32 height)
   at Avalonia.Win32.FramebufferManager.Lock()
   at Avalonia.Skia.FramebufferRenderTarget.CreateDrawingContext(IVisualBrushRenderer visualBrushRenderer)
   at Avalonia.Rendering.ImmediateRenderer.Paint(Rect rect)
   at Avalonia.Controls.TopLevel.HandlePaint(Rect rect)
   at Avalonia.Win32.WindowImpl.WndProc(IntPtr hWnd, UInt32 msg, IntPtr wParam, IntPtr lParam)
```
Reproduction: Run program and hit minimize-all windows on taskbar and then restore minimized application window.

**Comment by Silic0nS0ldier** [comment]
I've just encountered the same exception, but in a different way. I was also unable to reproduce the error using the described method. A theory is that different versions of Windows (and to a greater extent OSs) handle minimisation differently.
```
Unhandled Exception: System.ArgumentException: height is less than zero
   at Avalonia.Win32.WindowFramebuffer..ctor(IntPtr handle, Int32 width, Int32 height)
   at Avalonia.Win32.FramebufferManager.Lock()
   at Avalonia.Skia.FramebufferRenderTarget.CreateDrawingContext(IVisualBrushRenderer visualBrushRenderer)
   at Avalonia.Rendering.ImmediateRenderer.Paint(Rect rect)
   at Avalonia.Controls.TopLevel.HandlePaint(Rect rect)
   at Avalonia.Win32.WindowImpl.WndProc(IntPtr hWnd, UInt32 msg, IntPtr wParam, IntPtr lParam)
```
Reproduction: Under Windows 10, run program, and reduce height to the minimum.

**Comment by grokys** [comment]
I'm unable to reproduce this on `master` - is it still a problem?

**Comment by Gillibald** [comment]
I think I have fixed this here: #2120
